Invalid conversion from wxString to "wxChar *" in function getLinesUntil

When building wxmaxima 20.06.6 with GCC 10.1, we see errors when compiling `WXMformat.cpp` as follows:

```
[   26s] FAILED: src/CMakeFiles/wxmaxima.dir/WXMformat.cpp.o 
[   26s] /usr/bin/c++  -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -DwxDEBUG_LEVEL=0 -Isrc -isystem /usr/lib64/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -
O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG -pthre
ad -O2 -g -DNDEBUG   -fopenmp -std=gnu++14 -MD -MT src/CMakeFiles/wxmaxima.dir/WXMformat.cpp.o -MF src/CMakeFiles/wxmaxima.dir/WXMformat.cpp.o.d -o src/CMakeFiles/wxmaxima.d
ir/WXMformat.cpp.o -c ../src/WXMformat.cpp
[   26s] ../src/WXMformat.cpp: In function 'GroupCell* Format::TreeFromWXM(const wxArrayString&, Configuration**)':
[   26s] ../src/WXMformat.cpp:279:52: error: no match for call to '(const Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>) (const wxStrin
g&)'
[   26s]   279 |       line = getLinesUntil(Headers.GetEnd(headerId));
[   26s]       |                                                    ^
[   26s] ../src/WXMformat.cpp:224:30: note: candidate: 'Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>'
[   26s]   224 |   const auto getLinesUntil = [&wxmLine, end](const wxChar *tag) -> wxString
[   26s]       |                              ^
[   26s] ../src/WXMformat.cpp:224:30: note:   no known conversion for argument 1 from 'const wxString' to 'const wxChar*' {aka 'const wchar_t*'}
[   26s] ../src/WXMformat.cpp:286:52: error: no match for call to '(const Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>) (const wxStrin
g&)'
[   26s]   286 |       line = getLinesUntil(Headers.GetEnd(headerId));
[   26s]       |                                                    ^
[   26s] ../src/WXMformat.cpp:224:30: note: candidate: 'Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>'
[   26s]   224 |   const auto getLinesUntil = [&wxmLine, end](const wxChar *tag) -> wxString
[   26s]       |                              ^

```
and on and on.

This is fixed rather straightforwardly with this modification:
```diff
Index: wxmaxima-Version-20.06.6/src/WXMformat.cpp
===================================================================
--- wxmaxima-Version-20.06.6.orig/src/WXMformat.cpp
+++ wxmaxima-Version-20.06.6/src/WXMformat.cpp
@@ -221,7 +221,7 @@ GroupCell *TreeFromWXM(const wxArrayStri
 
   //! Consumes and concatenates lines until a closing tag is reached,
   //! consumes the tag and returns the line.
-  const auto getLinesUntil = [&wxmLine, end](const wxChar *tag) -> wxString
+  const auto getLinesUntil = [&wxmLine, end](const wxString &tag) -> wxString
   {
     wxString line;
     while (wxmLine != end)
```

I shall be happy to send you a PR if you agree this is the right fix.

Thank you for catching this.

That’s the right fix. Please PR it. It’s simple enough that you can do it directly in github web UI :)

> 3 juli 2020 kl. 2:09 em skrev badshah400 <notifications@github.com>:
> 
> ﻿
> When building wxmaxima 20.06.6 with GCC 10.1, we see errors when compiling WXMformat.cpp as follows:
> 
> [   26s] FAILED: src/CMakeFiles/wxmaxima.dir/WXMformat.cpp.o 
> [   26s] /usr/bin/c++  -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -DwxDEBUG_LEVEL=0 -Isrc -isystem /usr/lib64/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -
> O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG -pthre
> ad -O2 -g -DNDEBUG   -fopenmp -std=gnu++14 -MD -MT src/CMakeFiles/wxmaxima.dir/WXMformat.cpp.o -MF src/CMakeFiles/wxmaxima.dir/WXMformat.cpp.o.d -o src/CMakeFiles/wxmaxima.d
> ir/WXMformat.cpp.o -c ../src/WXMformat.cpp
> [   26s] ../src/WXMformat.cpp: In function 'GroupCell* Format::TreeFromWXM(const wxArrayString&, Configuration**)':
> [   26s] ../src/WXMformat.cpp:279:52: error: no match for call to '(const Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>) (const wxStrin
> g&)'
> [   26s]   279 |       line = getLinesUntil(Headers.GetEnd(headerId));
> [   26s]       |                                                    ^
> [   26s] ../src/WXMformat.cpp:224:30: note: candidate: 'Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>'
> [   26s]   224 |   const auto getLinesUntil = [&wxmLine, end](const wxChar *tag) -> wxString
> [   26s]       |                              ^
> [   26s] ../src/WXMformat.cpp:224:30: note:   no known conversion for argument 1 from 'const wxString' to 'const wxChar*' {aka 'const wchar_t*'}
> [   26s] ../src/WXMformat.cpp:286:52: error: no match for call to '(const Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>) (const wxStrin
> g&)'
> [   26s]   286 |       line = getLinesUntil(Headers.GetEnd(headerId));
> [   26s]       |                                                    ^
> [   26s] ../src/WXMformat.cpp:224:30: note: candidate: 'Format::TreeFromWXM(const wxArrayString&, Configuration**)::<lambda(const wxChar*)>'
> [   26s]   224 |   const auto getLinesUntil = [&wxmLine, end](const wxChar *tag) -> wxString
> [   26s]       |                              ^
> 
> and on and on.
> 
> This is fixed rather straightforwardly with this modification:
> 
> Index: wxmaxima-Version-20.06.6/src/WXMformat.cpp
> ===================================================================
> --- wxmaxima-Version-20.06.6.orig/src/WXMformat.cpp
> +++ wxmaxima-Version-20.06.6/src/WXMformat.cpp
> @@ -221,7 +221,7 @@ GroupCell *TreeFromWXM(const wxArrayStri
>  
>    //! Consumes and concatenates lines until a closing tag is reached,
>    //! consumes the tag and returns the line.
> -  const auto getLinesUntil = [&wxmLine, end](const wxChar *tag) -> wxString
> +  const auto getLinesUntil = [&wxmLine, end](const wxString &tag) -> wxString
>    {
>      wxString line;
>      while (wxmLine != end)
> I shall be happy to send you a PR if you agree this is the right fix.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

--- a/src/WXMformat.cpp
+++ b/src/WXMformat.cpp
@@ -221,7 +221,7 @@ GroupCell *TreeFromWXM(const wxArrayString &wxmLines, Configuration **config)
 
   //! Consumes and concatenates lines until a closing tag is reached,
   //! consumes the tag and returns the line.
-  const auto getLinesUntil = [&wxmLine, end](const wxChar *tag) -> wxString
+  const auto getLinesUntil = [&wxmLine, end](const wxString &tag) -> wxString
   {
     wxString line;
     while (wxmLine != end)

